sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/aio/client.py

```diff
@@ -0,0 +1,183 @@
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Any, Optional, Union
+    from azure.core.credentials import TokenCredential
+    from .. import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
+
+from azure.core.exceptions import HttpResponseError
+
+from .. import DecryptResult, EncryptResult, SignResult, VerifyResult, UnwrapKeyResult, WrapKeyResult
+from azure.keyvault.keys.models import Key
+from azure.keyvault.keys._shared import AsyncKeyVaultClientBase, parse_vault_id
+
+
+class CryptographyClient(AsyncKeyVaultClientBase):
+    """
+    Performs cryptographic operations using Azure Key Vault keys.
+
+    :param key:
+        Either a :class:`~azure.keyvault.keys.models.Key` instance as returned by
+        :func:`~azure.keyvault.keys.KeyClient.get_key`, or a string.
+        If a string, the value must be the full identifier of an Azure Key Vault key with a version.
+    :type key: str or :class:`~azure.keyvault.keys.models.Key`
+    :param credential: An object which can provide an access token for the vault, such as a credential from
+        :mod:`azure.identity`
+
+    Keyword arguments
+        - *api_version* - version of the Key Vault API to use. Defaults to the most recent.
+    """
+
+    def __init__(self, key: "Union[Key, str]", credential: "TokenCredential", **kwargs: "Any") -> None:
+        if isinstance(key, Key):
+            self._key = key
+            self._key_id = parse_vault_id(key.id)
+        elif isinstance(key, str):
+            self._key = None
+            self._key_id = parse_vault_id(key)
+            self._get_key_forbidden = None  # type: Optional[bool]
+        else:
+            raise ValueError("'key' must be a Key instance or a key ID string including a version")
+
+        if not self._key_id.version:
+            raise ValueError("'key' must include a version")
+
+        super(CryptographyClient, self).__init__(vault_url=self._key_id.vault_url, credential=credential, **kwargs)
+
+    @property
+    def key_id(self) -> str:
+        """
+        The full identifier of the client's key.
+
+        :rtype: str
+        """
+        return "/".join(self._key_id)
+
+    @property
+    async def key(self) -> "Optional[Key]":
+        """
+        The client's :class:`~azure.keyvault.keys.models.Key`.
+        Can be `None`, if the client lacks keys/get permission.
+
+        :rtype: :class:`~azure.keyvault.keys.models.Key`
+        """
+
+        if not (self._key or self._get_key_forbidden):
+            try:
+                self._key = await self._client.get_key(self._key_id.vault_url, self._key_id.name, self._key_id.version)
+            except HttpResponseError as ex:
+                self._get_key_forbidden = ex.status_code == 403
+        return self._key
+
+    async def encrypt(self, plaintext: bytes, algorithm: "EncryptionAlgorithm", **kwargs: "Any") -> EncryptResult:
```

Comment:
in other languages we moved the algorithm to the first parameter for all methods.